datasets/hr/public_officials/crawler.py

```diff
@@ -0,0 +1,126 @@
+         with organizational reference
+      * A start and/or end date with no position name or legal entity name results in an unspecified position
+         with no organizational reference. This is done for fidelity to the source.
+    """
+    legal_entity_name = data.pop("Pravna osoba u kojoj obnaša dužnost")
+    position_name = data.pop("dužnost")
+    start_date = data.pop("Datum početka obnašanja dužnosti")
+    end_date = data.pop("Datum kraja obnašanja dužnosti")
+    if any([legal_entity_name, position_name, start_date, end_date]):
+        legal_entity = make_legal_entity(context, legal_entity_name)
+        position_name = position_name or "Unspecified position"
+        position = h.make_position(
+            context,
+            position_name,
+            topics=None,
+            organization=legal_entity,
+        )
+
+        categorisation = categorise(context, position, is_pep=True)
+        occupancy = h.make_occupancy(
+            context,
+            person,
+            position,
+            no_end_implies_current=True,
+            categorisation=categorisation,
+            propagate_country=True,
+            start_date=convert_date(start_date),
+            end_date=convert_date(end_date),
+        )
+        if occupancy:
+            context.emit(position)
+            context.emit(occupancy)
+    context.audit_data(data)
+
+
+def make_person(context: Context, first_name: str, last_name: str) -> Entity:
+    person = context.make("Person")
+    person.id = context.make_slug(first_name, last_name)
+    person.add("firstName", first_name)
+    person.add("lastName", last_name)
+    person.add("country", "LT")
+    person.add("topics", "role.pep")
+    return person
+
+
+def validate_headers(context: Context, headers: list) -> None:
+    """Raises AssertionError if columns are missing or not in the expected order.
+    Logs extra columns. Note: This CSV has duplicate column names.
+    """
+    error_message = f"Unexpected column headers: {headers}"
+    assert headers[: len(FIELD_NAMES)] == FIELD_NAMES, error_message
+    if len(headers) > len(FIELD_NAMES):
+        context.log.warn(f"Unexpected column headers: {headers[:len(FIELD_NAMES):]}")
+
+
+def extract_dict_keys_by_prefix(
+    data: dict, key_names: list[str], prefix: str
+) -> dict[str:Any]:
+    """Pops dict keys in `key_names` if the key starts with `prefix`.
+    Returns a new dict and removes keys from old dict"""
+    return {
+        k.removeprefix(prefix): data.pop(k) for k in key_names if k.startswith(prefix)
+    }
+
+
+def crawl(context: Context):
+    """Fetches the current CSV file and crawls each row, making persons, legal entities, and positions"""
+    file_path = context.fetch_resource("daily_csv_release", context.data_url)
+    context.export_resource(file_path, CSV, title=context.SOURCE_TITLE)
+    with open(file_path, encoding="utf-8-sig") as fh:
+        reader = csv.DictReader(fh, fieldnames=FIELD_NAMES, delimiter=";")
```

Comment:
nice fix for their column header repetition. Could you also assert that the original headers are as we currently assume they are, e.g. just read the first row and close the file again and assert expected == actual? So we make sure we detect when they change their headers, breaking our assumption of the header meaning?

Reply:
Sure thing, just to be clear - if the expected columns are present and in order, but there are additional columns, we don't want to raise an error, right? We want to `log.warn` and move on. And if the expected columns are not present or out of order, only in that case will we raise the AssertionError and stop execution.

BTW looks like there is need to re-open the file - when passing `fieldnames` the DictReader assumes that the first line is data and reads it in.

Reply:
> if the expected columns are present and in order, but there are additional columns, we don't want to raise an error, right?

correct

> BTW looks like there is need to re-open the file - when passing fieldnames the DictReader assumes that the first line is data and reads it in.

oh of course!

Thanks!